gcp-jobs/filings-notebook-report/src/notebookreport/notebookreport.py

```diff
@@ -0,0 +1,190 @@
+    logging.info('Email with subject %s has been sent successfully!', subject)
+    server.quit()
+    if filename != '':
+        os.remove(os.path.join(os.getcwd(), r'data/')+filename)
+
+
+def processnotebooks(notebookdirectory, data_directory):
+    """Process Notebook."""
+    status = False
+    now = datetime.now()
+    warnings.filterwarnings('ignore', category=DeprecationWarning)
+
+    try:
+        retry_times = int(os.getenv('RETRY_TIMES', '1'))
+        retry_interval = int(os.getenv('RETRY_INTERVAL', '60'))
+        if notebookdirectory == 'monthly':
+            days = ast.literal_eval(os.getenv('MONTH_REPORT_DATES', ''))
+    except Exception:  # noqa: B902
+        logging.exception('Error processing notebook for %s', notebookdirectory)
+        send_email(notebookdirectory, data_directory, 'ERROR', traceback.format_exc())
+        return status
+
+    # For monthly tasks, we only run on the specified days
+    if notebookdirectory == 'daily' or (notebookdirectory == 'monthly' and now.day in days):
+        notebookdirectory = os.path.join(os.path.dirname(__file__), notebookdirectory)
+        logging.info('Processing: %s', notebookdirectory)
+
+        num_files = len(os.listdir(notebookdirectory))
+        file_processed = 0
+
+        for file in findfiles(notebookdirectory, '*.ipynb'):
+            file_processed += 1
+            note_book = os.path.basename(file)
+            for attempt in range(retry_times):
+                try:
+                    pm.execute_notebook(file, data_directory+'temp.ipynb', parameters=None)
+                    send_email(note_book, data_directory, '', '')
+                    os.remove(data_directory+'temp.ipynb')
+                    status = True
+                    break
+                except Exception:  # noqa: B902
+                    if attempt + 1 == retry_times:
+                        # If any errors occur with the notebook processing they will be logged to the log file
+                        logging.exception(
+                            'Error processing notebook %s at %s/%s try.', notebookdirectory, attempt + 1,
+                            retry_times)
+                        send_email(notebookdirectory, data_directory, 'ERROR', traceback.format_exc())
+                    else:
+                        # If any errors occur with the notebook processing they will be logged to the log file
+                        logging.exception('Error processing notebook %s at %s/%s try. '
+                                          'Sleeping for %s secs before next try', notebookdirectory, attempt + 1,
+                                          retry_times, retry_interval)
+                        time.sleep(retry_interval)
+                        continue
+            if not status and num_files == file_processed:
+                break
+    return status
+
+
+if __name__ == '__main__':
+    start_time = datetime.utcnow()
```

Comment:
can use .now(UTC) here and elsewhere (ruff may auto fix this for you)

Reply:
This was changed